docs/src/man/basics.md

```diff
@@ -1510,32 +1510,96 @@ julia> german[Not(5), r"S"]
+| `subset`     | Creates a new data frame.        | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `subset!`    | Modifies an existing data frame. | Retains both source and transformed columns. | Number of rows is determined by the transformation. |
+| `combine`    | Creates a new data frame.        | Retains only transformed columns.            | Number of rows is determined by the transformation. |
+
+### Constructing Transformation Pairs
+All of the functions above use the same syntax which is commonly
+`transformation_function(source_dataframe, transformation)`.
+The `transformation` argument is a `Pair` which defines the
+transformation to be applied to the `source_dataframe`,
+and it can take any of the forms listed below:
+
+- `source_column_selector => function`
+  - passes source column(s) to function
+  - automatically names the resulting column(s)
+- `source_column_selector => function => new_column_names`
+  - passes source column(s) to function
+  - names the resulting column(s) `new_column_names`
+    - *cannot be used with `subset` or `subset!`*
+- `source_column_selector => new_column_names`
+  - renames source column
+    - *may only select one column unless broadcasting*
+    - *may be used to split a column containing collections into multiple columns*
+    - *cannot be used with `subset` or `subset!`*
+- `source_column_selector`
+  - selects source column(s) without transforming them
+    - *often used with `select` or `select!` for isolating or moving columns*
+
+#### `source_column_selector`
+The most basic `source_column_selector` is a column name,
+but there are many more ways to select columns as explained
+in the first section of the [Indexing](@ref) API.
+
+#### `function`
+Here `function` is a function which operates on an entire data frame
+column passed as a vector.
+If you instead want to apply a function to each element in the column,
+then you can wrap your element-wise function in `ByRow` like
+`ByRow(my_elementwise_function)`.
+This will apply `my_elementwise_function` to every element in the column
+and then collect the results back into a vector.
+
+When multiple columns are selected by `source_column_selector`,
+the `function` will receive the columns as multiple positional arguments in the
+order they are selected like `f(column1, column2, column3)`.
+Slurping and splatting with `...` can be used to define a function which can
+accept any number of columns returned by `source_column_selector`.
+For example, the function `f(columns...) = max.(columns...)` will return a new column
+containing the largest element from each row for any number of input columns.
+
+#### `new_column_names`
+The results of the transformation will be placed into new columns
+with names defined by `new_column_names`.
+`new_column_names` may be a string, a symbol, a vector of strings,
+or a vector of symbols.
```

Comment:
or `AsTable`.

Reply:
Same here - `AsTable` can be discussed separately, but it is a common operation for splitting a column into multiple columns with automatic column naming:
```
julia> df = DataFrame(in=[(a=1,b=2), (a=3, b=4)])
2×1 DataFrame
 Row │ in
     │ NamedTup…
─────┼────────────────
   1 │ (a = 1, b = 2)
   2 │ (a = 3, b = 4)

julia> select(df, :in => AsTable)
2×2 DataFrame
 Row │ a      b
     │ Int64  Int64
─────┼──────────────
   1 │     1      2
   2 │     3      4

julia> select(df, :in => [:out_a, :out_b]) # you can also give the names of the columns explicitly
2×2 DataFrame
 Row │ out_a  out_b
     │ Int64  Int64
─────┼──────────────
   1 │     1      2
   2 │     3      4
```